src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -162,7 +161,7 @@
   (let [limit-clause     (if limit (format " LIMIT %s" limit) "")
         offset-clause    (if offset (format " OFFSET %s" offset) "")
         order-by-clause  (order-by->sql order_by)]
-    (format "SELECT paged_results.* FROM (%s) paged_results%s%s%s"
+    (format "SELECT paged_results.* FROM (%s %s%s%s) paged_results"
             sql order-by-clause limit-clause offset-clause)))
```

Comment:
So this is actually a candidate that I think would be more readable as honeysql, and we could use a cond-> to observe the limit/offset/order by decision tree /cc @mullr